File: src/main/java/com/example/dpop/orchestrator/api/v1/authentication/sms/EnrollStep.java
package com.example.dpop.orchestrator.api.v1.authentication.sms;

import com.example.dpop.auth_sms.EnrollmentRef;

import java.util.List;

public sealed interface EnrollStep {
    record NeedInput(List<String> missingFields)               implements EnrollStep {}
    record StartEnrollment(String phoneNumber)                  implements EnrollStep {}
    record ConfirmEnrollment(EnrollmentRef ref, String tan)    implements EnrollStep {}
    record ActivateMethod(EnrollmentRef ref)                    implements EnrollStep {}
}

File: src/main/kotlin/com/example/dpop/orchestrator/api/v1/authentication/sms/EnrollStep.kt
package com.example.dpop.orchestrator.api.v1.authentication.sms

import com.example.dpop.auth_sms.EnrollmentRef

sealed interface EnrollStep {
    @JvmRecord
    data class NeedInput(val missingFields: List<String>) : EnrollStep

    @JvmRecord
    data class StartEnrollment(val phoneNumber: String?) : EnrollStep

    @JvmRecord
    data class ConfirmEnrollment(val ref: EnrollmentRef, val tan: String?) : EnrollStep

    @JvmRecord
    data class ActivateMethod(val ref: EnrollmentRef) : EnrollStep
}
